### ISSUE TITLE
loss in vaebm

### ISSUE THREAD

**Issue by anitaweng** [NVlabs/VAEBM#2]
Hi, 
It is a nice work.
One question to ask, what `dvalue` does in this work?
In [train_VAEBM_distributed.py](https://github.com/NVlabs/VAEBM/blob/bee763c416d0febea1b13d12a5dec654e90d3b1e/train_VAEBM_distributed.py) 
`dvalue = model(neg_x_renorm) - log_p_total - log_pxgz`
`dvalue` then does backward, but both models are set `requires_grad` to `False` in line 213, 214.
Also, what are they correspond to respectively on the paper's loss function?
It would be many thank if anyone figure it out for me!

**Comment by arash-vahdat** [comment]
The part you are pointing out is where we are running Langevin dynamics to generate samples from the model which is shown in Eq. 2 in the [paper](https://arxiv.org/pdf/2010.00654.pdf). 

`dvalue` is the energy computed for the current sample which is shown in Eq. 4 in the paper. 

Note that Langevin dynamics generates samples given the current model by updating the samples using the gradient of energy w.r.t the samples. That's why `requires_grad` is `False` on the model parameters. The actual loss function that updates the energy function is computed at this [line](https://github.com/NVlabs/VAEBM/blob/bee763c416d0febea1b13d12a5dec654e90d3b1e/train_VAEBM_distributed.py#L290).
